src/sage/rings/complex_mpfr.pyx

```diff
@@ -1659,28 +1659,6 @@ cdef class ComplexNumber(sage.structure.element.FieldElement):
         mpfr_clear(right_nm)
         return x
 
-    def __rtruediv__(self, left):
-        r"""
-        Return the quotient of left with ``self``, that is:
-
```

Comment:
Note `__rtruediv__` is called explicitly in `RealIntervalFieldElement.__truediv__` (`src/sage/rings/real_mpfi.pyx`) and in `RealNumber.__truediv__` (`src/sage/rings/real_mpfr.pyx`).

I really fail to understand the logic behind those methods and why they sometimes use `__rtruediv__`; those methods are also deprecated (cf #15114) so maybe this doesn't matter at all.

Moreover, the examples in the doctests of the removed methods still work fine, so these methods seem really unnecessary.

Reply:
I think this is what happened. Only `__truediv__` is special for Cython code (with `c_api_binop_methods=True`, which is the case both before Cython 3 and for our settings in Cython 3), and both `x.__truediv__` and `x.__rtruediv__` are available because they are wrappers of the actual division code. Here, `__rtruediv__` is simply a normal method that shadows the name of the special wrapper, so `x.__rtruediv__` is still available after removing this code.

`RealIntervalFieldElement` and `RealNumber` only call `__rtruediv__` on `Element` and `RealIntervalFieldElement`, so they are irrelevant here.